core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -39,12 +45,13 @@ public OutputStream compress(OutputStream os) throws IOException {
     }
   };
 
-  private static final ConcurrentMap<String, Decompressor> decompressors =
+  private static final ConcurrentMap<String, DecompressorInfo> decompressors =
       initializeDefaultDecompressors();
 
   /**
    * Represents a message compressor.
    */
+  @ExperimentalApi
```

Comment:
I'm just worried about maintenance. As soon as we add a method and forget to add the annotation ... what does that mean for the method?  Inconsistency here may be confusing to the user.

With that said, I'm fine deferring to your judgement here :)